cli/src/commands/extensions/mod.rs

```diff
@@ -108,6 +115,41 @@ async fn handle_install_extension(path: &str) -> CommandResult {
     let extension_path = PathBuf::from(path);
     let extension = Extension::try_from(extension_path)?;
 
+    let permissions = extension.permissions();
+
+    // Attempt to construct a `PermissionsOptions` from the `Permissions`
+    // object in order to validate the permissions.
+    let _ = PermissionsOptions::try_from(permissions)?;
+
+    if !accept_permissions && !extension.permissions().is_allow_none() {
```

Comment:
Wonder if an early return would be better here? It's kinda far down in the method but it would allow us to remove the indentation of a pretty big block of code.

Otherwise my best recommendation would probably be to move that code to a separate method?